atsign-foundation/at_c
# feat: update mbedtls to v3.5.1 in atchops & atclient

closes #85 



**- What I did**

- updated MbedTLS dependencies in atchops and atclient to latest release v3.5.1
- made fix to breaking change that v3.4 -> v3.5 introduced.

**- How I did it**

- v3.4 -> v3.5 introduced a new bug with using the MBEDTLS_MD enums... enum of '4' indicated a different SHA algorithm... the correct new enum is '5'. This PR fixes that
- conversation with an MbedTLS member can be found here: https://github.com/Mbed-TLS/mbedtls/issues/8737

**- How to verify it**

- Tests still passing
- Did manual test of pkam_authenticate. See below

```

 [INFO] atclient | atclient_connection_connect: 0. Successfully connected to root
 [INFO] connection |    SENT: "smoothalligator
"
 [INFO] connection |    RECV: "245b44d4-a4bd-5f33-b077-c559f956486a.swarm0001.atsign.zone:1722"
 [INFO] atclient | atclient_connection_connect: 0. Successfully connected to secondary
 [INFO] connection |    SENT: "from:@smoothalligator
"
 [INFO] connection |    RECV: "data:_96ee0fbc-045b-4b1d-bccc-313c6a900559@smoothalligator:b24d493c-10f0-4097-a968-53211836d24d"
rsa check privkey: 0
mbedtls_ctr_drbg_seed: 0
mbedtls_rsa_pkcs1_sign: 0
5c 67 e8 2d 66 4d 3f 17 35 64 60 1c fb 92 4e ac 89 12 cb 42 60 df 18 a5 f5 15 68 e5 3e 22 11 e8 cb 24 11 77 ea d2 cf db 37 65 15 90 ee 81 d4 6d 6d 5b 5f 47 c8 72 d5 06 67 53 80 87 a1 c6 92 17 1a 42 cd 61 68 68 7b 01 bb cf fe 1a ea ec fd ab ec 3f 2a c9 b3 23 a2 7c d3 49 09 7a 4a b0 92 35 8b 51 e4 32 38 64 84 c4 d4 c7 3a 44 e2 3d 44 5f cf c9 07 fb 78 f9 bd c7 61 d4 d1 e2 d7 dc 2a e4 71 a0 09 ab b9 38 06 02 1a a4 9d 12 f8 6d 29 64 da 7e ee 1b 94 ee 30 6b b6 21 ab bc cc 00 2b d2 4e 7d ee 98 a8 ec 1a e3 5d 16 ff 68 18 8b 5e 69 22 9a c2 05 0f d4 ee 1a d5 34 6d e2 17 94 4c 8a 7f e9 11 f3 0f 0f 1f 49 0e 00 8b 09 2d 73 2c 84 f5 4c 02 af 35 39 7b 78 d3 cf 2c 6e d0 45 c8 ac 38 8a 00 49 6f 1a 33 90 36 b4 13 9e 5c fb 06 ad 2d ad 8c 10 35 76 3b c4 d1 bd 5b 04 e6 6d 54 8d 
atchops_base64_encode: 0
 [INFO] connection |    SENT: "pkam:XGfoLWZNPxc1ZGAc+5JOrIkSy0Jg3xil9RVo5T4iEejLJBF36tLP2zdlFZDugdRtbVtfR8hy1QZnU4CHocaSFxpCzWFoaHsBu8/+Gurs/avsPyrJsyOifNNJCXpKsJI1i1HkMjhkhMTUxzpE4j1EX8/JB/t4+b3HYdTR4tfcKuRxoAmruTgGAhqknRL4bSlk2n7uG5TuMGu2Iau8zAAr0k597pio7BrjXRb/aBiLXmkimsIFD9TuGtU0beIXlEyKf+kR8w8PH0kOAIsJLXMshPVMAq81OXt4088sbtBFyKw4igBJbxozkDa0E55c+watLa2MEDV2O8TRvVsE5m1UjQ==
"
 [INFO] connection |    RECV: "data:success"
```

**- Description for the changelog**
